plugins/base/src/main/kotlin/signatures/JvmSignatureUtils.kt

```diff
@@ -133,6 +135,10 @@ interface JvmSignatureUtils {
                 it.dri == DRI("kotlin", "Deprecated")
                         || it.dri == DRI("java.lang", "Deprecated")
             } == true) setOf(TextStyle.Strikethrough) else emptySet()
+
+    infix fun DFunction.uses(t: DTypeParameter): Boolean =
+        t.dri in (listOfNotNull(receiver?.type?.drisOfAllNestedBounds, receiver?.dri) +
+                parameters.flatMap { listOf(it.dri) + it.type.drisOfAllNestedBounds })
```

Comment:
This is wrong. The list you are looking at has inferred type `List<Any>` since it contains both `DRI`s and `List<DRI>`s.

Reply:
Done